editor/src/components/canvas/controls/grid-controls.tsx

```diff
@@ -962,11 +964,346 @@ export const GridControls = controlForStrategyMemoized<GridControlsProps>(({ tar
             }}
           />
         ) : null}
+        <AbsoluteDistanceIndicators targetRootCell={targetRootCell} />
       </CanvasOffsetWrapper>
     </React.Fragment>
   )
 })
 
+const MIN_INDICATORS_DISTANCE = 32 // px
+
+const AbsoluteDistanceIndicators = React.memo(
+  (props: { targetRootCell: GridCellCoordinates | null }) => {
+    const colorTheme = useColorTheme()
+
+    const cellFrame = useEditorState(
+      Substores.metadata,
+      (store) => {
+        if (store.editor.selectedViews.length !== 1) {
+          return null
+        }
+
+        const meta = MetadataUtils.findElementByElementPath(
+          store.editor.jsxMetadata,
+          store.editor.selectedViews[0],
+        )
+        if (!MetadataUtils.isPositionAbsolute(meta)) {
+          return null
+        }
+
+        return nullIfInfinity(meta?.globalFrame)
+      },
+      'AbsoluteDistanceIndicators cellFrame',
+    )
+    const canvasScale = useEditorState(
+      Substores.canvasOffset,
+      (store) => store.editor.canvas.scale,
+      'AbsoluteDistanceIndicators canvasScale',
+    )
+
+    const canvasOffset = useEditorState(
+      Substores.canvasOffset,
+      (store) => store.editor.canvas.roundedCanvasOffset,
+      'AbsoluteDistanceIndicators canvasOffset',
+    )
+
+    const targetCellBoundingBox = React.useMemo(() => {
+      if (props.targetRootCell == null) {
+        return null
+      }
+      const element = document.querySelector(
+        `[data-grid-row="${props.targetRootCell.row}"]` +
+          `[data-grid-column="${props.targetRootCell.column}"]`,
+      )
+      const boundingBox = element?.getBoundingClientRect()
+      if (boundingBox == null) {
+        return null
+      }
+
+      const canvasOrigin = windowToCanvasCoordinates(
+        canvasScale,
+        canvasOffset,
+        windowPoint({ x: boundingBox.left, y: boundingBox.top }),
+      ).canvasPositionRounded
+      const canvasRect = canvasRectangle({
+        x: canvasOrigin.x,
+        y: canvasOrigin.y,
+        width: boundingBox.width * canvasScale,
+        height: boundingBox.height * canvasScale,
+      })
+
+      return canvasRect
+    }, [props.targetRootCell, canvasScale, canvasOffset])
+
+    const distanceTop =
+      targetCellBoundingBox == null || cellFrame == null ? 0 : cellFrame.y - targetCellBoundingBox.y
+
+    const distanceLeft =
+      targetCellBoundingBox == null || cellFrame == null ? 0 : cellFrame.x - targetCellBoundingBox.x
+
+    const positioning = React.useMemo(() => {
+      if (cellFrame == null || targetCellBoundingBox == null) {
+        return null
+      }
+
+      function position(
```

Comment:
I would suggest positioning them in the middle of the axes (like the blue indicators):
<img width="284" alt="image" src="https://github.com/user-attachments/assets/bff7a8ef-7dee-4c28-9372-4470a460b123">
vs
<img width="277" alt="Monosnap flawless-santanaraptor - Utopia 2024-09-09 17-01-54" src="https://github.com/user-attachments/assets/e663ff12-e348-41d3-917d-5b3421959f44">

Reply:
I don't agree on this one: the positioning at the origin on the element is to better clarify this is an offset (relative to the `0;0` coord of the bounding rectangle)

Reply:
but it's not changing this meaning, since the "left" indicator will still be from the left side and the "top" will still be from the top side (they will just be positioned in the middle of the side itself).
but it's very minor, so np